Create GameField class

VladiusVostokus opened the pull request:
Has methods:
- updateGameField() which create field with 4 walls and space inside.
- getCell()/setCell() to get and set cell in some string value.
- showGameField() to display gamefield in console as a strings.
- getHeight()/getWidth() used in tests to check correct behaviour after several updates.

In tests I check the state of gamefield after updates, whether get\set methods work properly and whether the gamefield array is filled with the correct values.

VladiusVostokus committed:
Add first version class of gamefield

This is 2-d array, with 2 methods:
- updateGameField: it creates the initial gamefield with borders. First
  of all, it fills by copies of rows of ' ', and then its "borders" are
  filled with the '#'
- showGameField: it dislays gamefield array as a string, so there are no
  brackets or commas on the screen.
Now, this is looks like "empty box", but it will be changed in the
future.

VladiusVostokus committed:
Write test for gamefield and ifx problem

Tests check the size of gamefiedl, and they shown that if update
gamefield twise, it will be bigger by its initial height. It's because
array initializing happens while class instance creating, so rows were
pushed to previous array, not new. So I move initializing to
updateGameField() method.

VladiusVostokus committed:
Replace 9 with this.width - 1

I write 9 here earlier, because the width of gamefiedl in index.ts was
10, but width can be other, and hardcoded 9 can cause problems.

VladiusVostokus committed:
Test for getCell() and setCell()

Test whether setCell() set a certain value to the cell and whether
getCell() can return this value.

VladiusVostokus committed:
Add test for gamefiedl wall and space value

Check if wall cells are equal to '#' after creating the gamefield and
check if cells inside equal to ' ' after creating, so walls and space
created correctly.

stbestichhh commented on src/gamefield.ts:
```diff
@@ -0,0 +1,48 @@
+export class GameField {
+    symbol: string;
+    width: number;
+    height: number;
+    row: string[];
+    gameField!: string[][];
```
have to be private

VladiusVostokus replied:
Done

stbestichhh commented on src/gamefield.ts:
```diff
@@ -0,0 +1,48 @@
+    }
+
+    updateGameField() {
+        this.gameField = [];
```
has to be set in constructor

VladiusVostokus replied:
I move it from constructor to this method, because in the other way the height will increase with each update, so it needs to be in update method and constructor both?

stbestichhh replied:
You have to initialise game field in constructor instead of using not null "!" operator.

VladiusVostokus replied:
Done

stbestichhh commented on src/gamefield.ts:
```diff
@@ -0,0 +1,48 @@
+        console.log(field); 
+    }
+
+    getCell(row: number, col: number) {
+        return this.gameField[row][col];
+    }
+
+    setCell(row: number, col: number, value: string) {
+        this.gameField[row][col] = value;
+    }
+
+    getWidth() {
+        return this.gameField[0].length;
+    }
+
+    getHeight() {
+        return this.gameField.length;
+    }
```
use getters and setters instead of this

stbestichhh commented on src/index.ts:
```diff
@@ -1 +1,6 @@
-console.log('Hello, World!');
+import { GameField } from './gamefield';
+
+const gameField = new GameField(' ', 10, 10)
```
use constants or proccess.argv to set size

stbestichhh replied:
"Magic numbers"

VladiusVostokus replied:
Done

stbestichhh commented on test/gamefield.spec.ts:
```diff
@@ -0,0 +1,46 @@
+import { GameField } from '../src/gamefield';
+
+const gameField = new GameField(' ', 10, 10);
```
better to create a class instance before each tests instead of global variable

VladiusVostokus replied:
Done

VladiusVostokus committed:
Make fiedls private

VladiusVostokus committed:
Make initialization of gameField in constructor

VladiusVostokus committed:
constans intead of magic numbers

VladiusVostokus committed:
create a class instance before each tests

VladiusVostokus committed:
Remove hardcoded 9 from row in updare method

Put actual width - 1 instead, so there shouldn't be size issues.

VladiusVostokus committed:
Symbol as field of class

I think it's more suitable to make space symbol as private field instead
of passing it to the constructor as an argument. Also it was a
bit strange: why choose the space when creating class instance?

VladiusVostokus committed:
Get for wall and space

I decided to make wall as class fiedl, so I have 2 wall and space as
fields and did getter for them to use in tests, this decrease hardcode
and also prevent problems when change values of wall or space.

VladiusVostokus commented:
I made some changes here:
- fix issues you writed
- fix problem with with index inside class
- make wall and space as private field
- make get for wall and space to use in tests
